Tests/CommonHelpers/Acts/Tests/CommonHelpers/FloatComparisons.hpp

```diff
@@ -160,7 +160,10 @@ predicate_result matrixCompare(const Eigen::DenseBase<Derived1>& val,
 //        only accept them. Does someone know a clean way to do that in C++?
 //
 template <typename Container,
-          typename Enable = typename Container::const_iterator>
+          typename CbeginDefined =
+              std::decay_t<decltype(*cbegin(std::declval<Container>()))>,
```

Comment:
Two things surprise me here:

- I would expect `cbegin` to be only accessible via `std::cbegin`, not in isolation.
- If this is only used for SFINAE, I do not understand why the `std::decay_t` is needed.